rero_ils/modules/items/serializers/csv_ext.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2022 RERO
+# Copyright (C) 2022 UCLouvain
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Item serializers."""
+
+from csv import QUOTE_ALL, DictWriter
+
+from flask import current_app, request, stream_with_context
+from invenio_i18n.ext import current_i18n
+from invenio_records_rest.serializers.csv import CSVSerializer, Line
+
+from rero_ils.modules.item_types.api import ItemTypesSearch
+from rero_ils.modules.libraries.api import LibrariesSearch
+from rero_ils.modules.locations.api import LocationsSearch
+from rero_ils.modules.serializers import CachedDataSerializerMixin
+from rero_ils.utils import get_i18n_supported_languages
+
+from .collector_ext import CollecterExt as Collecter
+
+
+class ItemExtCSVSerializer(CSVSerializer, CachedDataSerializerMixin):
+    """Serialize item search for csv."""
+
+    def serialize_search(self, pid_fetcher, search_result, links=None,
```

Comment:
I have a suggestion regarding this class. but not sure it works.

First of all, add an attribute `collector` to  `ItemCSVSerializer` class
```python

from .collector import Collecter

class ItemCSVSerializer(CSVSerializer, CachedDataSerializerMixin):
    """Serialize item search for csv."""

    collector = Collecter
```
Then, modify code in `ItemCSVSerializer.ItemCSVSerializer()` to use new attribute
``` python
...
for pids, batch_results in self.collector.batch(results=search_result):
  # get documents
  documents = self.collector.get_documents_by_item_pids(
      item_pids=pids, language=language)
  # get loans
  items_stats = self.collector.get_loans_by_item_pids(item_pids=pids)
  for hit in batch_results:
      csv_data = self.collector.get_item_data(hit)
      # _process_item_types_libs_locs(self, csv_data)
      _process_item_types_libs_locs(csv_data)
      self.collector.append_document_data(csv_data, documents)
      self.collector.append_local_fields(csv_data)
      # update csv data with loan
      self.collector.append_loan_data(hit, csv_data, items_stats)
      self.collector.append_issue_data(hit, csv_data)
      # write csv data
      data = self.process_dict(dictionary=csv_data)
      writer.writerow(data)
      yield line.read()
...
```
Finally, you can do this in `ItemExtCSVSerializer` class:
```python
...
from .collector_ext import CollecterExt

from .collector import Collecter

class ItemExtCSVSerializer(ItemCSVSerializer, CachedDataSerializerMixin):
    """Serialize item search for csv."""

    collector = CollecterExt
```